### ISSUE TITLE
Partition table with foreign key as key problems

### ISSUE THREAD

**Issue by slapshin** [SectorLabs/django-postgres-extra#123]
Hi! Thank you for the useful tool!
I have some problems/questions about generating migration for partitioned models.

For example we have model:
```
class MyModel(PostgresPartitionedModel):
    class PartitioningMeta:
        method = PostgresPartitioningMethod.RANGE
        key = ["foreign_field"]

    title = models.CharField(...)
    foreign_field = models.ForeignKey(...)
```

migration looks like:
```
    operations = [
         psqlextra.backend.migrations.operations.create_partitioned_model.PostgresCreatePartitionedModel(...),
         psqlextra.backend.migrations.operations.add_default_partition.PostgresAddDefaultPartition(...),
         migrations.AddField(
            model_name='MyModel',
            name='foreign_field',
        ),

```
So, firstly table with "simple" fields is created and after that foreign field will be added to the table. But if this field is used as a partition key we will have the error:
"Model 'MyModel' is not properly configured to be partitioned. Field 'foreign_field' in partitioning key ['foreign_field'] is not a valid field on 'MyModel'". Because the model doesn't "see" this field on executing the PostgresCreatePartitionedModel operation. Why relations are adding after table creation? Maybe it is better to combine this?

**Comment by opqpop** [comment]
bump, was anyone able to find a solution for this?

**Comment by davmlaw** [comment]
Create the migration, then manually move the FK creation line to above the ```partitioning_options```

**Created with pgmakemigrations**

```
        psqlextra.backend.migrations.operations.create_partitioned_model.PostgresCreatePartitionedModel(
            name='MyModel',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('title', models.CharField()),
                # Needs to go here
            ],
            partitioning_options={
                'method': psqlextra.types.PostgresPartitioningMethod['LIST'],
                'key': ['foreign_field'],
            },
        # Skipped
        ),
        # Need to move this into fields above partitioning options
        migrations.AddField(
            model_name='mymodel',
            name='foreign_field',
            field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='+', to='foreign_model'),
        ),
```

** Manually Modified **

```
        psqlextra.backend.migrations.operations.create_partitioned_model.PostgresCreatePartitionedModel(
            name='MyModel',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('title', models.CharField()),
                ('foreign_field', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='+', to='foreign_model')),
            ],
            partitioning_options={
                'method': psqlextra.types.PostgresPartitioningMethod['LIST'],
                'key': ['foreign_field'],
            },
        # Skipped
        ),
```

**Comment by Photonios** [comment]
I think this has been solved in v2.0.4rc1 as a result of PR #150. Please give it a shot :)
